test/boost/stall_free_test.cc

```diff
@@ -62,7 +64,7 @@ SEASTAR_THREAD_TEST_CASE(test_clear_gently_trivial_unique_ptr) {
     std::unique_ptr<int> p = std::make_unique<int>(0);
 
     utils::clear_gently(p).get();
-    BOOST_CHECK(p);
+    BOOST_REQUIRE(!p);
```

Comment:
checking `!p` seems like indeed the right thing, how did checking the opposite work before this patch, and why did we expect the opposite to be true?